python/desc/sims/GCRCatSimInterface/InstanceCatalogWriter.py

```diff
@@ -133,21 +147,49 @@ def write_catalog(self, obsHistID, out_dir='.', fov=2):
             # Creating empty knots component
             subprocess.check_call('cd %(out_dir)s; touch %(knots_name)s' % locals(), shell=True)
 
-        bulge_db = bulgeDESCQAObject(self.descqa_catalog)
-        bulge_db.field_ra = self.protoDC2_ra
-        bulge_db.field_dec = self.protoDC2_dec
-        cat = self.instcats.DESCQACat(bulge_db, obs_metadata=obs_md,
-                                      cannot_be_null=['hasBulge'])
-        cat.write_catalog(os.path.join(out_dir, gal_name), chunk_size=100000,
-                          write_header=False)
-
-        disk_db = diskDESCQAObject(self.descqa_catalog)
-        disk_db.field_ra = self.protoDC2_ra
-        disk_db.field_dec = self.protoDC2_dec
-        cat = self.instcats.DESCQACat(disk_db, obs_metadata=obs_md,
-                                      cannot_be_null=['hasDisk'])
-        cat.write_catalog(os.path.join(out_dir, gal_name), chunk_size=100000,
-                          write_mode='a', write_header=False)
+        if self.sprinkler is False:
```

Comment:
We have not thought about how the sprinkler will interact with the knots catalog.

The knots code is meant to add more interesting structure to galaxy disks.  There is currently a non-zero probability that it will try to add knots to a galaxy that has been replaced by the sprinkler.  In that case, the knots will be added based on wrong assumptions about the size and SED of the galaxy.  Probably, the safest thing to do at this point is add code to prevent the knots catalogs from touching the sprinkled galaxies.  Maybe we can add a column to all `DESCQAObject`s named `sprinkled`.  This will be an int that defaults to 0, but gets set to 1 by the sprinkler.  We can then modify the knots catalog to ignore any galaxy with `sprinkled==1`.

Thoughts?

Reply:
@EiffL This is what I was referring to

Reply:
Turns out, we won't be using knots in Run 1.1

@jbkalmbach: If you need someone to just approve this, you can ping me.

I will go back to minding my own business

Reply:
@danielsf  oh ok, I see, yes it makes a lot of sense to not add knots to the sprinkled galaxies. We can add open an issue on this after the code gets merged if we don't want to delay the sprinkler at this point